test/integration/garbagecollector/garbage_collector_test.go

```diff
@@ -453,18 +453,15 @@ func setupRCsPods(t *testing.T, gc *garbagecollector.GarbageCollector, clientSet
 		pod := newPod(podName, namespace, []metav1.OwnerReference{{UID: rc.ObjectMeta.UID, Name: rc.ObjectMeta.Name}})
 		createdPod, err := podClient.Create(context.TODO(), pod, metav1.CreateOptions{})
 		if err != nil {
-			t.Fatalf("Failed to create Pod: %v", err)
+			t.Errorf("Failed to create Pod: %v", err)
 		}
 		podUIDs = append(podUIDs, createdPod.ObjectMeta.UID)
 	}
 	orphan := false
 	switch {
-	case options.OrphanDependents == nil && options.PropagationPolicy == nil && len(initialFinalizers) == 0:
```

Comment:
I'm guessing this was dropped, but was it long enough ago that we can do this according to our deprecation policy?

Reply:
What is the policy?
I have made changes for these since `options.OrphanDependents` is deprecated,

Reply:
https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions - if this was dropped 1.17 or later, we need to keep it; if it was dropped 1.16 or earlier, we can drop it

Reply:
It is deprecated in 1.7 [link](https://v1-16.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/)

Reply:
But maybe this part needs a seperate PR since there are other methods which use still use it like `getOrphanOptions()`, `getNonOrphanOptions()`, [here](https://github.com/kubernetes/kubernetes/blob/ec705e6142a31583e2b1a904fa9c347a6b751ad0/test/integration/garbagecollector/garbage_collector_test.go#L543), ...

Reply:
Looking through kubernetes/kubernetes CHANGELOGs I can't find anything that explicitly calls this as having been deprecated.  There is an entry in 1.6 that says its _going_ to be deprecated in 1.7, but I find no such announcement in later releases (ref: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.6.md#garbage-collector)

So yeah I'd rather see this reverted and a separate issue opened up for its removal

Reply:
@liggitt do you know who could answer whether this has actually been deprecated?

Reply:
cc @caesarxuchao @deads2k

Reply:
in general, the correctness tests should remain present as long as the field does, whether it is marked deprecated or not.